pkg/sql/opt/norm/fold_constants_funcs.go

```diff
@@ -408,6 +408,34 @@ func (c *CustomFuncs) foldOIDFamilyCast(
 		default:
 			return nil, false, nil
 		}
+	case oid.T_regproc:
+		cDatum, err := eval.PerformCast(c.f.ctx, c.f.evalCtx, datum, typ)
```

Comment:
I'm not convinced we need to use `PerformCast` here. If we use it, we'll need some more checks to ensure that the cast is valid, e.g. use `cast.ValidCast`, otherwise we'll be attempting to fold expressions like `12.3::FLOAT::REGPROC` which I think would fail.

Can we instead use similar logic as in the `case oid.T_oid:` at the top of the switch?

Reply:
i think we should be able to use `PerformCast` as long as we add similar cases like the ones we have in the Regtype case